Modules/Microsoft365DSC/DSCResources/MSFT_TeamsGroupPolicyAssignment/MSFT_TeamsGroupPolicyAssignment.psm1

```diff
@@ -0,0 +1,433 @@
+            ApplicationId         = $ApplicationId
+            TenantId              = $TenantId
+            CertificateThumbprint = $CertificateThumbprint
+        }
+    }
+    catch
+    {
+        New-M365DSCLogEntry -Message 'Error retrieving data:' `
+            -Exception $_ `
+            -Source $($MyInvocation.MyCommand.Source) `
+            -TenantId $TenantId `
+            -Credential $Credential
+
+        return $nullReturn
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Id,
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $GroupDisplayname,
+
+        [Parameter()]
+        [System.String]
+        $GroupId,
+
+        [Parameter()]
+        [System.String]
+        $PolicyType,
+
+        [Parameter()]
+        [System.String]
+        $PolicyName,
+
+        [Parameter()]
+        [System.String]
+        $Priority,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint
+    )
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace 'MSFT_', ''
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftTeams' -InboundParameters $PSBoundParameters
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+
+    #check policyname
+    $command = "get-cs" + $PolicyType
+    $Policys = Invoke-Expression -Command $command -ErrorAction SilentlyContinue
+    $policymatch = $false
+    if($null -ne $Policys){
+        Foreach($policy in $Policys.Identity){
+            $match = "^Tag:" + $PolicyName + "$"
+            if($policy -match $match){
+                $policymatch = $true
+            }
+        }
+    }
+    if($null -eq $Policys -or $policymatch -eq $false){
+        Write-Verbose -Message "No PolicyType found for $PolicyType"
+        return
+    }
+
+    #get groupid
+    if($GroupId.Length -eq 0)
+    {
+        $Group = Find-CsGroup -SearchQuery $GroupDisplayname
+        if($group.Length -gt 1)
+        {
+            Write-Verbose -Message "Found $($group.Length) groups with the name $GroupDisplayname"
+            $Group = $Group | Where-Object { $_.DisplayName -eq $GroupDisplayname }
```

Comment:
Just wondering: On line 201 you search on the GroupDisplayName. If that results in two groups with that specific display name, when you filter again for that same GroupDisplayName on this line: Doesn't this result in the same set of groups?

Reply:
Yes, you are right, in this case is no possible to find the right group. The second Filter is useful in case there are groups named "AllUser", "AllUserUS", "AllUserDE" and so on, with a Filter "AllUser" i will get it all, with the second filter -eq i will get only "AllUser". The -SearchQuery is like a -match.